crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -428,7 +437,7 @@ pub fn infix_operand_type_rules(
             if let TypeBinding::Bound(binding) = &*int.borrow() {
                 return infix_operand_type_rules(binding, op, other, errors);
             }
-            if other.try_bind_to_polymorphic_int(int, is_const, op.span).is_ok() {
+            if other.try_bind_to_polymorphic_int(int, is_const, op.span).is_ok() || other == &Type::Error {
```

Comment:
This extra clause was added because it was possible to get an error for `Types in a binary operation should match, but found Error and Field` before.